.github/workflows/fetch-otp-data.yml

```diff
@@ -0,0 +1,74 @@
+name: 'Fetch and process data to build OTP graph'
+
+on:
+  workflow_call:
+  workflow_dispatch:
+    inputs:
+      with_elevation:
+          description: 'Include elevation file in OTP data source'
+          default: false
+          type: boolean
+      local_download:
+          description: 'Output otp_dir artifact, default uploads to S3 bucket'
+          default: false
+          type: boolean
+
+env:
+  CAC_OTP_DATA_BUCKET: 'cleanair-otp-data'
+
+jobs:
+  fetch-data:
+    name: Fetch and process GTFS, OSM and elevation data
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout GitHub Repo
+        uses: actions/checkout@v4
+
+      - uses: aws-actions/configure-aws-credentials@v4
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: ${{ secrets.AWS_REGION }}
+
+      - name: Fetch GTFS zip files
+        working-directory: deployment/dataproc/gtfs-feed-fetcher
+        run: docker compose run --rm gtfs-feed-fetcher ./fetch_cac_feeds.py
+
+        # TODO 1397: Edit SEPTA bus routes.txt files to bring back to OTPv1 support,
+        # Confirm no errors in .html files, run feedvalidator script
+
+      - name: Extend GTFS feed date range
+        working-directory: deployment/dataproc/gtfs-feed-fetcher
+        run: docker compose run --rm gtfs-feed-fetcher ./extend_effective_dates.py --feeds=patco.zip,septa_bus.zip,septa_rail.zip,dart.zip
+
+      - name: Copy GTFS file to otp_data
+        run: cp deployment/dataproc/gtfs-feed-fetcher/gtfs/{patco_extended.zip,septa_bus_extended.zip,septa_rail_extended.zip,dart_extended.zip,nj_bus.zip,nj_rail.zip} ./otp_data/
+
+      - name: Fetch OSM files
+        working-directory: deployment/dataproc/osm
+        run: docker compose run --rm osm ./download-osm-data.sh
+
+      - name: Process OSM files
+        working-directory: deployment/dataproc/osm
+        run: docker compose run --rm osm ./combine-and-clip.sh
+
+      - name: Copy OSM .pbf file to otp_data
+        run: |
+          sudo chmod 644 deployment/dataproc/osm/cac.pbf
+          cp deployment/dataproc/osm/cac.pbf ./otp_data/
+
+      - name: Upload latest GTFS and OSM files to S3
+        if:  ${{ !inputs.local_download }}
+        run: aws s3 sync "otp_data" "s3://$CAC_OTP_DATA_BUCKET/"
+
+      - name: Fetch elevation file from S3
+        if:  ${{ !inputs.local_download && inputs.with_elevation }}
+        run: aws s3 cp "s3://$CAC_OTP_DATA_BUCKET/cac.tif" "otp_data/cac.tif"
+
+      - name: Output otp_dir artifact for local download
```

Comment:
With the elevation file, the output can be ~4+GB, which may cost a fair bit of the [GitHub Action artifact size quota](https://docs.github.com/en/billing/managing-billing-for-your-products/managing-billing-for-github-actions/about-billing-for-github-actions#calculating-minute-and-storage-spending). Should be used very judiciously.

Reply:
Great point, I'm wondering if I just go ahead and take these conditional steps out? It would be easy enough for someone to run this job and then just pull the S3 bucket data locally if they need it, which I can write an extra line in the README changes in PR https://github.com/azavea/cac-tripplanner/pull/1407 for that scenario